auth/server/base/src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -1162,6 +1165,14 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                     AuthenticationConfigurationCallback authenticationConfigurationCallback = (AuthenticationConfigurationCallback) callback;
                     saslSkipCertificateVerification = authenticationConfigurationCallback.getSaslSkipCertificateVerification();
                     handleOne(callbacks, idx + 1);
+                } else if (callback instanceof RequestInformationCallback) {
+                    HashMap<String, Object> props = ((RequestInformationCallback) callback).getProperties();
+                    Attributes runtimeAttributes = new MapAttributes();
+                    for (Map.Entry<String, Object> entry : props.entrySet()) {
+                        runtimeAttributes.addFirst(entry.getKey(), entry.getValue().toString());
```

Comment:
@darranl @fjuma Updated this PR to use a map of <String, Function> as `properties` in the callback. I am not sure if it is okay to add **all** of the properties to the runtimeAttributes and not just `Request-URI` ? I wanted to introduce a `public static final String REQUEST_URI` somewhere in this repo, but I did not find a good place without introducing new dependencies or classes with constants, and this is a maintenance branch also. But maybe putting all of the properties from the callback  into the  `runtimeAttributes` indiscriminately makes sense now.

Also, this is relying on the `toString()` method being implemented well on the `entry.getValue()`.

Let me know what you think.  Thanks!

Reply:
Since this is more generic now, adding all of the properties to the `runtimeAttributes` makes sense to me.

For the `REQUEST_URI` constant, maybe that could be added to the `SetRequestInformationCallbackMechanismFactory` and then it could be used from WildFly Core?

Reply:
@fjuma Thanks, it is a good idea, though it might make make an unneeded precedence that the REQUEST_URI should be one of the properties or that all possible properties should  be placed there.
@darranl WDYT ?

Reply:
I think I would be tempted to try and keep the list of properties out of Elytron for now but in WildFly Core it should define Functions for all properties.

Late on I think custom implementations of the functions could be added to extract Cookie information etc so may be good to control the default behaviour over there?

Reply:
@darranl Okay sounds good to me, and that is the current state of these PRs. Request-URI and its function is currently defined in wildfly-core as seen in the [PR ](https://github.com/wildfly/wildfly-core/pull/5779) and if any other functions will be added in the future, those can be added to the same place in wildfly-core. Please check that your approval on wildfly-core still holds as the code has been changed since your last approval. Thank you!

@fjuma Let us know if you agree and if so, can you please approve this PR and the wildfly-core [PR](https://github.com/wildfly/wildfly-core/pull/5779)? Thank you!